components/suggestion/suggestion_box.test.jsx

```diff
@@ -127,4 +128,35 @@ describe('components/SuggestionBox', () => {
         instance.handleCompositionUpdate({data: '@저'});
         expect(instance.handlePretextChanged).toBeCalledWith('@저');
     });
+
+    test('should reset selection after provider.handlePretextChanged is handled', () => {
+        const userid1 = {id: 'userid1', username: 'user', first_name: 'a', last_name: 'b', nickname: 'c'};
+        const userid2 = {id: 'userid2', username: 'user2', first_name: 'd', last_name: 'e', nickname: 'f'};
+        const userid3 = {id: 'userid3', username: 'other', first_name: 'X', last_name: 'Y', nickname: 'Z'};
+
+        const baseParams = {
+            currentUserId: 'userid1',
+            profilesInChannel: [userid1, userid2, userid3],
+            autocompleteUsersInChannel: jest.fn().mockResolvedValue(false),
+        };
+        const provider = new AtMentionProvider(baseParams);
+        const props = {
+            ...baseProps,
+            providers: [provider],
+        };
+        const wrapper = shallow(
+            <SuggestionBox
+                {...props}
+            />
+        );
+        const instance = wrapper.instance();
+
+        expect(wrapper.state('selection')).toEqual('');
+
+        instance.nonDebouncedPretextChanged('hello world @');
+        expect(wrapper.state('selection')).toEqual('@other');
+
```

Comment:
Maybe wen can be more exhaustive here with the test adding:

```suggestion

        instance.nonDebouncedPretextChanged('hello world @u');
        expect(wrapper.state('selection')).toEqual('@user');
        
        instance.nonDebouncedPretextChanged('hello world @');
        expect(wrapper.state('selection')).toEqual('@user');

```

Reply:
Great suggestion, updated test 👍